src/components/ArticleDetails.jsx

```diff
@@ -0,0 +1,49 @@
+import React, { useEffect, useState } from "react";
+import getSingleArticle from "../utils/getSingleArticle";
+import { useParams } from "react-router-dom";
+import SingleArticle from "./SingleArticle";
+
+const ArticleDetails = () => {
+	const { article_id } = useParams();
+	const [singleArticle, setSingleArticle] = useState([]);
+	const [isError, setIsError] = useState(false);
+	const [isLoading, setIsLoading] = useState(false);
+
+	useEffect(() => {
+		getSingleArticle(article_id)
+			.then((article) => {
+				setSingleArticle(article);
+				console.log(article, "<-use effect article");
+				setIsLoading(false);
+			})
+			.catch((err) => {
+				setIsError(true);
+				setIsLoading(true);
```

Comment:
if there's an error we can still set is loading to be false as the page is no longer loading - we've received a response even if it's not the one we might have wanted!